### PR TITLE
completed ticket 10

### PR DESCRIPTION
**Pull request by manheicatherine**
I completed ticket 10, please check! Thank you very much!

### CODE REVIEW

**Review by alexdevdavis — CHANGES_REQUESTED** [review]
Catherine, these should be queries, not sent on the body. 
- check the notes for how to use req.queries, I'd suggest, then refactor this thing, and submit another PR

File: models/articles.models.js [hunk 1]
```diff
@@ -1,20 +1,51 @@
...
+          return Promise.reject({ status: 404, msg: "404 Not Found" });
+        } else {
+          const deleteUnuseItem = rows.map((each) => {
+            delete each.body;
+            return each;
+          });
+          return deleteUnuseItem;
+        }
+      });
+  } else if (topic === undefined) {
```
Comment (itskatherine) [inline models/articles.models.js]: This method's working, if you wanted to improve the code and avoid repetition you could add the topic to the query if it is specified (i.e. have a query string that has the WHERE topic = $1 appended to it, depending on whether or not it has been defined.)🙂

File: models/articles.models.js [hunk 2]
```diff
@@ -1,20 +1,51 @@
...
+        `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
       FROM articles LEFT JOIN comments
       ON articles.article_id = comments.article_id
-      GROUP BY articles.article_id ORDER BY articles.created_at DESC;`
-    )
-    .then(({ rows }) => {
-      const deleteUnuseItem = rows.map((each) => {
-        delete each.body;
-        return each;
+      GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`
```
Comment (itskatherine) [inline models/articles.models.js]: Here (and on line 15) we leave ourselves open to SQL injection. We can't use the $1 syntax to protect from this in this instance, so we need another way to make sure the user input here is safe.
We need to include a list of valid (greenlisted) sort_by options, and order options, and check the user input is included in those lists before proceeding with the query. We need to provide the same list for the topics as well (though this list can be populated by the fetchTopics function you've already written).

File: __tests__/app.test.js [hunk 3]
```diff
@@ -30,19 +30,73 @@ describe("API Testing", () => {
...
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSorted({ descending: false });
+          expect(body.articles.length).toBe(11);
+        });
+    });
+
+    test("returns an array of articles when passing non-existing topic", () => {
+      return request(app)
+        .get("/api/articles?howareyou=imfine")
```
Comment (itskatherine) [inline __tests__/app.test.js]: This should instead be a 400 bad request, we're sending through invalid info to make our request

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Catherine, great work so far, however some changes are needed to protect from SQL injection in our query, have a look at my comments and please send another PR once you've addressed them 🙂

File: __tests__/app.test.js [hunk 4]
```diff
@@ -30,17 +30,106 @@ describe("API Testing", () => {
...
+        .get("/api/articles?sort_by=title")
         .expect(200)
         .then(({ body }) => {
-          const objOfArr = body.articles;
-          expect(objOfArr.length).toBe(12);
-          expect(objOfArr[0].created_at).toBe("2020-11-03T09:12:00.000Z");
-          expect(objOfArr[objOfArr.length - 1].created_at).toBe(
-            "2020-01-07T14:08:00.000Z"
-          );
+          console.log(body)
```
Comment (HeatherEver) [inline __tests__/app.test.js]: remove console logs :)

File: models/articles.models.js [hunk 5]
```diff
@@ -1,20 +1,96 @@
...
+  //         return Promise.reject({ status: 404, msg: "404 Not Found" });
+  //       } else {
+  //         const deleteUnuseItem = rows.map((each) => {
+  //           delete each.body;
+  //           return each;
+  //         });
+  //         return deleteUnuseItem;
+  //       }
+  //     });
+  // }
```
Comment (HeatherEver) [inline models/articles.models.js]: commented out code should be deleted before committing :)

**Review by HeatherEver — APPROVED** [review]
Good work here! Merge away

File: models/articles.models.js (hunk 5 above)
Comment (manheicatherine) [reply models/articles.models.js]: no problem! Thank you very much!

**Pull request merged**